### PR TITLE
Reminder and recall

### PR DESCRIPTION
**Pull request by teryhill**
reminder and recall #581

### CODE REVIEW

File: interface/main/messages/print_postcards.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  "p.fname, p.mname, p.lname, p.pubpid, p.DOB, " .
+  "p.street, p.city, p.state, p.postal_code, p.pid " .
+  "FROM patient_data AS p " .
+  "WHERE p.pid = ? LIMIT 1", array($pid));
+
+# sprintf to print data
+$text = sprintf("  %s %s\n  %s\n  %s %s %s\n ", $patdata['fname'], $patdata['lname'], $patdata['street'], $patdata['city'], $patdata['state'], $patdata['postal_code']);
+
+$postcard_message = "It's time to get your EYES checked!
+Please call our office to schedule\nyour eye exam  at (413) 276-4543.
```
Comment (aethelwulffe) [inline interface/main/messages/print_postcards.php]: This hard-coded stuff needs to come from the facility data or a custom template and configuration.

File: interface/main/messages/save.php [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+        if ($logged_in) {
+            $token      = $logged_in['token'];
+            $practice   = $hb->practice->sync($token);
+            $token      = $logged_in['token'];
+
+            $response   = $hb->practice->sync($token);
+            $campaigns  = $hb->campaign->events($token);
+            $response   = $hb->events->generate($token,$campaigns['events']);
+
+            $response['success'] = "OK BABY!";
```
Comment (aethelwulffe) [inline interface/main/messages/save.php]: That is the required response?

File: interface/main/messages/print_postcards.php (hunk 1 above)
Comment (teryhill) [inline interface/main/messages/print_postcards.php]: We need to work on the postcards. They will be configurable from the MedExbank server like the SMS, Email and AVM messages. We were working on these first and then get to the other features.

**Comment by aethelwulffe** [comment]
While this stuff could probably make it to a module, it is currently wrapped up in the general mashup of the communication tools.  If we push it to interface and libraries, we should perhaps do just a bit more work on it first.  The post-cards and other bits should be customizable pretty early on (lest we forget), or perhaps we can just move a piece or two into the already existing directories under /sites as "Examples" that don't have a real clinic's info in them.  I will see if I can help with this a little.

**Comment by teryhill** [comment]
I will be cutting down the assets by removing the items that we are not using.

**Comment by teryhill** [comment]
@aethelwulffe I am working on the post cards.  What do you think will be the best way of getting the information in the body of the card. The facility information is there.

postcard

**Comment by teryhill** [comment]
@aethelwulffe medexbank,com is the service that sends the SMS, AVM and emails.

**Comment by aethelwulffe** [comment]
I have been getting the test SMS and email just fine.  I assume my responses have shown up.

**Comment by teryhill** [comment]
Yes the confirm by SMS and the confirm by e-mail you sent showed up just fine. I have a commit I need to make for some small fixes. Right now working on a way to stop the automatic refresh on tracker so notes can be entered.

**Comment by aethelwulffe** [comment]
I am pretty sure this piece is going to take a lot of security review etc... and I am gonna need to put some real work into the reviews...  Do we have a clinic that is going to be using this as a test/production in the immediate future?

**Comment by teryhill** [comment]
I will bringing up a customer in the next week or so. It is currently running on Dr Ray's system.

**Comment by aethelwulffe** [comment]
Cool.
Speaking (out of context) about Dr. Ray, could you push the eye form to the new forms repo?

**Comment by teryhill** [comment]
That will take some work. I can put it back on the list, If it is wanted?

**Comment by aethelwulffe** [comment]
IfNot upgrade changes look correct.

**Comment by teryhill** [comment]
conflicts resolved. I tested the script on my system which already had the medex options installed.

File: interface/main/messages/save.php [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+        $data['sender_name']    = $user_data['fname']. " " .$user_data['lname'];
+        $data['sender_email']   = $facility['email'];
+        $data['callerid']       = $facility['phone'];
+        $data['MedEx']          = "1";
+        $data['ipaddress']      = $_SERVER['REMOTE_ADDR'];
+
+        $prefix ='http://';
+        if ($_SERVER["SSL_TLS_SNI"]) $prefix = "https://";
+        $data['website_url']    = $prefix.$_SERVER['HTTP_HOST'].$web_root;
+        $practice_logo = "$OE_SITE_DIR/images/practice_logo.gif";
```
Comment (teryhill) [inline interface/main/messages/save.php]: What is the valid value?

File: interface/main/messages/save.php [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+                        <li> ".xlt('Activate your subscription plan')."</li>
+                        <li> ".xlt('Contact support if your server\'s IP address is dynamic')."</li>
+                    </ul>
+                </div>
+                ";
+            echo json_encode($response_prob);
+        }
+                    //then redirect user to preferences with a success message!
+    } else {
+        echo xlt("Sorry you are not privileged enough...");
```
Comment (teryhill) [inline interface/main/messages/save.php]: Changed

File: interface/main/messages/js/reminder_appts.js [hunk 5]
```diff
@@ -0,0 +1,464 @@
...
+ * @link http:Librehealth.io
+ */
+
+
+var labels=[];
+var postcards=[];
+var show_just;
+
+$(function() {
+  $( "#datepicker1" ).datepicker({ dateFormat: "yy-mm-dd" });
```
Comment (teryhill) [inline interface/main/messages/js/reminder_appts.js]: no it is not the same, also the one in patient portal is the same they both will need to be changes at some point in time

File: interface/main/messages/save.php [hunk 6]
```diff
@@ -109,10 +111,10 @@
 
             $runQuery ="select * from facility order by name";
             $fetch = sqlStatement($runQuery);
-            while ($frow = sqlFetchArray($fetch)) { $facilities[] = $frow; }
+            while ( $frow = sqlFetchArray( $fetch ) ) { $facilities[] = $frow['id']; }
```
Comment (aethelwulffe) [inline interface/main/messages/save.php]: Ouch.  Guess this doesn't happen much.  Is there a way to do full end-to-end validation via their API to make sure all the pieces work?

File: library/MedEx/API.php [hunk 7]
```diff
@@ -2058,7 +2069,8 @@ public function login() {
         $this->curl->setData(array(
```
Comment (aethelwulffe) [inline library/MedEx/API.php]: any particular special cURL configuration stuff needed server-side?
I suppose the install needs to check for:
extension=php_curl.dll in phpinfo

Comment (teryhill) [inline library/MedEx/API.php]: Did not have to do anything to my standard windows set for my test box or the user that is getting the code

File: interface/main/messages/save.php (hunk 6 above)
Comment (teryhill) [inline interface/main/messages/save.php]: It is working on my test system and has been for a while

File: library/MedEx/API.php (hunk 7 above)
Comment (aethelwulffe) [inline library/MedEx/API.php]: Finally scrolled down past the giant word-pressy things on their site to find out how much it costs.  Something tells me that our outfit isn't going to be able to afford the $62,400 a year subscription fees for text messaging...

**Comment by teryhill** [comment]
not up to you. the cost is 99.00 per month per provider

On Tuesday, July 11, 2017, 3:42:40 PM EDT, Art Eaton <notifications@github.com> wrote:


@aethelwulffe commented on this pull request.

In library/MedEx/API.php:
> @@ -2058,7 +2069,8 @@ public function login() {
         $this->curl->setData(array(

Finally scrolled down past the giant word-pressy things on their site to find out how much it costs. Something tells me that our outfit isn't going to be able to afford the $62,400 a year subscription fees for text messaging...

**Comment by aethelwulffe** [comment]
Well, 45 counselors x12x100.00 makes it basically an impossible dream.

**Comment by teryhill** [comment]
start with the busiest ones or just a couple and see if it improves your customer communication. I am sure we can work some kind of a deal for you 8^)

**Comment by aethelwulffe** [comment]
They all busy, and most share clients with a different program, and there is plenty of turnover that would make things crazy.  60K is two salaries, and $100 a month is more than they pay for internet and two phones, so unless it comes out to being something more like the cost of a phone line, this is waaaay out of reach, even at .  HIPAA fax service is only 20 a month for the whole business.  eRX is 50.

**Comment by teryhill** [comment]
Sorry Art. Most of my practices are 1 to 5 providers and the cost is manageable.

**Comment by aethelwulffe** [comment]
Why close this?  We can't afford it, but I am sure that there are plenty of 90210's that may want it.  It does need to be an add-on, but this is an operable connector.

**Comment by tmccormi** [comment]
Agreed.  Make it an add-on module.  It a good option for many practices

**Comment by teryhill** [comment]
Sorry the close was an oops

**Comment by teryhill** [comment]
I will add some controls to turn it off but the code is very heavily embedded in Patient Tracker beyond me to make it a module I do not understand hoe to accomplish it.

**Comment by tmccormi** [comment]
Good reason for a hack-a-thon ...

**Comment by teryhill** [comment]
I will pull the code until that happens it that is what y'all want. I have it working for the version my clients are on and since it appears to not be a fit with LibreHealth EHR. it is just taking up space.

**Comment by teryhill** [comment]
Just an FYI for you guys

	https://www.remindercall.com/pricing-2/  
	Example: 20pts/day x 20d/month = 400 events.  
With a 60 second phone call, and one SMS with a reply, you are at 1600 units which is $127.50/month.

**Comment by aethelwulffe** [comment]
Heh, Terry...I guess we are just waaaaay too darn poor.  Do they take 
bottlecaps?

Honestly, I will look at stuff and see just how much they would actually 
be sending.  On a volume rate, we are OK.  On a per-provider 
rate...hahahahahahahahaha...:-D

**Comment by aethelwulffe** [comment]
Any volunteers to take this over and create a "module" out of it?

**Pull request closed without merging**